workflows/kallisto-quant/run-kallisto.nf

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env nextflow
+nextflow.enable.dsl=2
+
+// run parameters
+params.ref_dir = 's3://nextflow-ccdl-data/reference/homo_sapiens/ensembl-100'
+params.index_dir = 'kallisto_index'
+params.index_name = 'cdna_k31'
+params.annotation_dir = 'annotation'
+params.t2g = 'Homo_sapiens.ensembl.100.tx2gene.tsv'
+params.mitolist = 'Homo_sapiens.ensembl.100.mitogenes.txt'
+params.barcodes = 's3://nextflow-ccdl-data/reference/10X/barcodes/3M-february-2018.txt' // 10X v3 barcodes
+
+params.sample_dir = 's3://ccdl-scpca-data/raw/green_adam'
+params.sample_ids = "834,905_3" //comma separated list to be parsed into a list
+params.outdir = 's3://nextflow-ccdl-results/scpca/kallisto-quant'
+
+// build full paths
+params.index_path = "${params.ref_dir}/${params.index_dir}/${params.index_name}"
+params.t2g_path = "${params.ref_dir}/${params.annotation_dir}/${params.t2g}"
+params.mito_path = "${params.ref_dir}/${params.annotation_dir}/${params.mitolist}"
+
+
+process kallisto_bus{
+  container 'quay.io/biocontainers/kallisto:0.46.2--h4f7b962_1'
+  label 'cpus_8'
+  tag "${id}-${index}"
+  publishDir "${params.outdir}"
+  input:
+    tuple val(id), path(read1), path(read2)
+    path index
+  output:
+    path run_dir
+  script:
+    run_dir = "${id}-${index}"
+    // interleave read1 & read2 files
```

Comment:
Tell me more?

Reply:
It is not actually interleaving the data, just the file paths.
Took a while to figure this out, but this takes two lists, say [1, 2, 3] and [10, 20, 30], and makes the transpose matrix (if they had been combined): [[1,10], [2,20], [3,30]]. I then `flatten()` this to get the files in the order needed for kallisto, which expects R1 and R2 pairs in order. Finally `join()` with spaces between...

Reply:
I might suggest

```suggestion
    // interleave read1 & read2 file paths
```

in that case because I definitely read it as interleaving the data!